### ISSUE TITLE
fasta2speclib feature requests

### ISSUE THREAD

**Issue by JB91451** [compomics/ms2pip_c#188]
Dear ms2pip developers,

Thank you for the new version of your tool. It works really well.

I have rather a question than an issue: Is there any way to include semi-specific or even unspecific enzymatically cleaved peptides in the fasta2speclib tool? 
In the earlier versions of ms2pip I _bypassed_ fasta2speclib for this purpose by using a script to generate the peptide list via generate-peptides from crux (btw. this did allow to filter peptides also by m/z which was a huge advantage -> I agree with issue #187 that this would be a nice option in the future) and modifying this list to obtain suitable peprec files including a 'max variable modifications per peptide' option. However, it would be nice if it is possible to use the _offical_ way. 

Best regards,
Juergen

**Comment by RalfG** [comment]
Dear Juergen, thanks for your interest in fasta2speclib! We are currently working on optimizing fasta2speclib, so we should be able to implement your suggestions in the next version.

**Comment by RalfG** [comment]
fasta2speclib feature requests:
- [x] Semi-specific cleavage
- [x] Unspecific cleavage
- [x] Min and max precursor m/z
- [x] Max variable modifications per peptide

**Comment by RalfG** [comment]
@JB91451, @willems533,

All requested features are now available in MS²PIP v3.11.0. Please let us know if everything works as expected. If you would run into any issues, let us know and we'll be happy to help.

Best,
Ralf

**Comment by JB91451** [comment]
Dear Ralf,

Thank you and the entire MS2PIP team for this really fast implementation! I will test it as soon as possible. Unfortunately, this week I am buisy with some lab courses.

Best,
Juergen

**Comment by patrick-willems** [comment]
Hey Ralf,

Thanks! I will also test it soon in one of the coming days,

Patrick

**Comment by JB91451** [comment]
Hi Ralf,

I tried the new features today and they seem to work. However, when I set more than one cpu in the config file, the process seems to get stuck in the multiprocessing step at the very beginning. For the tests I used a VM-Server with win10 and python 3.7 (I can not change the version there due to access restrictions). Even with only one cpu the process seems to stop at a step "generated new fontManager".
When I run the very same config file from my desktop computer (python 3.10) multiprocessing works as expected but I have to reduce the chunk-size. 

On the computer where the process does not finish, I also have to use a virtual environment in order to call fasta2speclib while on my desktop I installed ms2pip globally and could run both commands directly.

Best,
Juergen
